app/controllers/join_team_requests_controller.rb

```diff
@@ -1,98 +1,105 @@
 class JoinTeamRequestsController < ApplicationController
-  def action_allowed?
-    current_role_name.eql?("Student")
-  end
+	# decide if the controller is accessisable to the user
+	before_action :set_event, only => [ :show, :edit, :update, :destroy, :decline]
 
-  def index
-    @join_team_requests = JoinTeamRequest.all
+	def action_allowed?
+	  current_role_name.eql?("Student")
+	end
 
-    respond_to do |format|
-      format.html # index.html.erb
-      format.xml  { render xml: @join_team_requests }
-    end
-  end
+	def index
+	  @join_team_requests = JoinTeamRequest.all
+	  render
+	end
 
-  def show
-    @join_team_request = JoinTeamRequest.find(params[:id])
-    respond_to do |format|
-      format.html # show.html.erb
-      format.xml  { render xml: @join_team_request }
-    end
-  end
+	def show
+	  render
+	end
 
-  def new
-    @join_team_request = JoinTeamRequest.new
-    respond_to do |format|
-      format.html # new.html.erb
-      format.xml  { render xml: @join_team_request }
-    end
-  end
+	def new
+	  @join_team_request = JoinTeamRequest.new
+	  render
+	end
 
-  def edit
-    @join_team_request = JoinTeamRequest.find(params[:id])
-  end
+	def edit
+	end
+
+	# create a new join team request entry for join_team_request table and add it to the table
+	def create
```

Comment:
the indentation might made it confusing, I manually corrected the wrong identation style.
But we did refactored the react_to_format and checkteam_flash out of this method.